### PR TITLE
fix: update to recent standards

### PR DESCRIPTION
**Pull request by cklanac**
- update gitignore with common node assets
- update npm packages
- add fix for `mocha --exit` in v4x
- server.js and test-server.js: add `use strict`, fix formatting, use promises, cleanup

Request review @benjaminEwhite

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,9 +1,12 @@
 <!DOCTYPE html>
 <html>
+
 <head>
 	<title></title>
 </head>
+
 <body>
-<p>hello world</p>
+	<div>hello world</div>
```
Comment (benjaminEwhite) [inline public/index.html]: Why `<div>` over `<p>`?

**Review by benjaminEwhite — APPROVED** [review]
Just one question but otherwise LGTM 👍

File: public/index.html (hunk 1 above)
Comment (cklanac) [reply public/index.html]: Oh, that's nothing. I started to expand the page to make a simple ajax/fetch call back to the API, but then bailed on the idea.

Comment (cklanac) [reply public/index.html]: I'll revert the change for consistency

**Pull request merged**